python/ray/autoscaler/_private/_azure/node_provider.py

```diff
@@ -136,9 +136,11 @@ def non_terminated_nodes(self, tag_filters):
         nodes() must be called again to refresh results.
 
         Examples:
-            >>> provider.non_terminated_nodes({TAG_RAY_NODE_KIND: "worker"})
+            >>> provider.non_terminated_nodes({TAG_RAY_NODE_KIND: "worker", TAG_RAY_CLUSTER_NAME: "cluster_name"})
             ["node-1", "node-2"]
         """
+        tag_cluster_name = {TAG_RAY_CLUSTER_NAME: self.cluster_name}
+        tag_filters.update(tag_cluster_name)
```

Comment:
wouldn't this be better to move under _get_filtered_nodes? also I would just add a direct check for that tag vs. updating the tag_filters dict to avoid undesired side effects to the dictionary reference.

Reply:
Thanks. I have moved it under `_get_filtered_nodes`, and also make further modifications to avoid undesired side effects to the dictionary reference.